src/MCPClient/lib/clientScripts/archivematicaFITS.py

```diff
@@ -33,93 +32,91 @@
-import databaseInterface
+from executeOrRunSubProcess import executeOrRun
 
-# initialize Django (required for Django 1.7)
 import django
 django.setup()
+# dashboard
+from main.models import FPCommandOutput
+
+
+logger = get_script_logger("archivematica.mcp.client.FITS")
 
-databaseInterface.printSQL = False
-excludeJhoveProperties = True
 formats = []
 FITSNS = "{http://hul.harvard.edu/ois/xml/ns/fits/fits_output}"
 
 
-def excludeJhoveProperties(fits):
-    """Exclude <properties> from <fits><toolOutput><tool name="Jhove" version="1.5"><repInfo> because that field contains unnecessary excess data and the key data are covered by output from other FITS tools."""
-    formatValidation = None
-
+def exclude_jhove_properties(fits):
+    """
+    Exclude <properties> from "/fits/toolOutput/tool[name=Jhove]/repInfo"
+    because that field contains unnecessary excess data and the key data are
+    covered by output from other FITS tools.
+    """
+    format_validation = None
     tools = getTagged(getTagged(fits, FITSNS + "toolOutput")[0], FITSNS + "tool")
     for tool in tools:
         if tool.get("name") == "Jhove":
-            formatValidation = tool
+            format_validation = tool
             break
-    if formatValidation == None:
+    if format_validation == None:
         return fits
-    repInfo = getTagged(formatValidation, "repInfo")[0]
+    repInfo = getTagged(format_validation, "repInfo")[0]
     properties = getTagged(repInfo, "properties")
-
     if len(properties):
         repInfo.remove(properties[0])
     return fits
 
 
-if __name__ == '__main__':
-    logger = get_script_logger("archivematica.mcp.client.FITS")
-
-    global exitCode
-    exitCode = 0
-    target = sys.argv[1]
-    XMLfile = sys.argv[2]
-    date = sys.argv[3]
-    eventUUID = sys.argv[4]
-    fileUUID  = sys.argv[5]
-    fileGrpUse = sys.argv[6]
-
-    if fileGrpUse in ["DSPACEMETS", "maildirFile"]:
-        print "file's fileGrpUse in exclusion list, skipping"
-        exit(0)
+def main(target, xml_file, date, event_uuid, file_uuid, file_grpuse):
+    """
+    Note: xml_file, date and event_uuid are not being used.
+    """
+    if file_grpuse in ('DSPACEMETS', 'maildirFile'):
+        logger.error("File's fileGrpUse in exclusion list, skipping")
+        return 0
 
-    sql = """SELECT fileUUID FROM FPCommandOutput WHERE fileUUID = %s;"""
-    if len(databaseInterface.queryAllSQL(sql, (fileUUID,))):
-        print >>sys.stderr, "Warning: Fits has already run on this file. Not running again."
-        exit(0)
+    if not FPCommandOutput.objects.filter(file=file_uuid).count():
+        logger.error('Warning: Fits has already run on this file. Not running again.')
+        return 0
 
-    tempFile="/tmp/" + uuid.uuid4().__str__()
-
-    command = "fits.sh -i \"" + escapeForCommand(target) + "\" -o \"" + tempFile + "\""
+    _, temp_file = tempfile.mkstemp()
+    args = 'fits.sh -i {} -o {}'.format(target, temp_file)
```

Comment:
If this is built as an array, we won't have to worry about shlex.split, and executeOrRun knows how to handle it.  Suggest `args = ['fits.sh', '-i', target, '-o', temp_file]`